src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -148,47 +170,24 @@ private static void setupLogging(Settings settings, Environment environment) {
         return InternalSettingsPreparer.prepareSettings(EMPTY_SETTINGS, true);
     }
 
-    /**
-     * hook for JSVC
-     */
-    public void init(String[] args) throws Exception {
-        Tuple<Settings, Environment> tuple = initialSettings();
-        Settings settings = tuple.v1();
-        Environment environment = tuple.v2();
-        setupLogging(settings, environment);
-        setup(true, settings, environment);
-    }
-
-    /**
-     * hook for JSVC
-     */
-    public void start() {
+    private void start() {
         node.start();
+        keepAliveThread.start();
     }
 
-    /**
-     * hook for JSVC
-     */
-    public void stop() {
-       destroy();
-    }
-
-
-    /**
-     * hook for JSVC
-     */
-    public void destroy() {
-        node.close();
-    }
-
-    public static void close(String[] args) {
-        bootstrap.destroy();
-        keepAliveLatch.countDown();
+    private void stop() {
+        try {
+            if (node != null) {
```

Comment:
I didn't make bootstrap Closeable because i didnt want close to be a public method. The class must be public so its main method can be run, but otherwise a lot of this stuff is still crazy. like even the static INSTANCE would be great to eliminate.

We could factor out the java code to be different... I can do it here if we really need but I would love to defer that.